includes/class-wc-payments-account.php

```diff
@@ -1027,20 +872,16 @@ public function maybe_handle_onboarding() {
 						$args['is_progressive_onboarding'] = $this->is_progressive_onboarding_in_progress() ?? false;
 					}
 
-					$this->redirect_to_account_link( $args );
+					$this->redirect_service->redirect_to_account_link( $args );
 				}
 
-				$this->redirect_to_login();
+				// Clear account transient when generating Stripe dashboard's login link.
+				$this->clear_cache();
```

Comment:
extracted this call from `redirect_to_login` function to remove dependency from WC_Payments_Account